java/org.travlyn.server/src/main/java/org/travlyn/server/externalapi/access/DBpediaStopRequest.java

```diff
@@ -21,23 +48,61 @@ public DBpediaStopRequest(String query) {
 
     @Override
     public Stop getResult() {
+        //check for manual name setting
+        String manualQuery = this.checkForManualName(query);
+        if (manualQuery != null){
+            query = manualQuery;
+        }
+
         Set<Pair<String, String>> params = getDefaultHeaders(query);
         params.add(new Pair<>("property", "dbo:abstract"));
         params.add(new Pair<>("property", "dbo:thumbnail"));
 
         String result = executeRequest(params);
-        JsonObject englishContent = gson.fromJson(result, JsonObject.class).getAsJsonObject("results").
-                getAsJsonArray("bindings").get(0).getAsJsonObject();
+        JsonArray resultArray;
+        JsonObject englishContent = null;
+        try {
+             resultArray = gson.fromJson(result, JsonObject.class).getAsJsonObject("results").
+                    getAsJsonArray("bindings");
+        }catch (JsonSyntaxException syntaxException){
+            //quota limit reached...exclude stop TODO
+            return null;
```

Comment:
What about this case? Maybe throw an exception instead of just returning null? At least internally (server side)